siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+    # to do: make so that selection changes on first click
+    if "selected" not in streamlit.session_state:
+        streamlit.session_state['selected'] = []
+    if "expanded" not in streamlit.session_state:
+        streamlit.session_state['expanded'] = []
+
+    selected = tree_select(logs_and_reports,
+                           expand_on_click=True,
+                           checked=streamlit.session_state['selected'],
+                           expanded=streamlit.session_state['expanded'],
+                           only_leaf_checkboxes=True)
+
+    if len(selected['checked']) == 0:
+        streamlit.session_state['selected'] = []
+    if len(selected["checked"]) == 1:
+        streamlit.session_state['selected'] = selected["checked"]
+    if len(selected["checked"]) > 1:
+        for x in selected["checked"]:
+            if x != streamlit.session_state['selected'][0]:
+                newly_selected = x
+                break
+        streamlit.session_state['selected'] = [newly_selected]
+        streamlit.session_state['expanded'] = selected["expanded"]
+        streamlit.session_state['right after rerun'] = True
+        streamlit.experimental_rerun()
+
+    if streamlit.session_state.selected != []:
+        return True
+    return False
+
+
+def show_metrics_of_log_or_report(chip, step, index):
+    """
+    Displays the metrics that are included in each file.
+
+    Args:
+        chip (Chip) : the chip object that contains the schema read from
+        step (string) : step of node
+        index (string) : index of node
+    """
+    if 'selected' in streamlit.session_state and \
+       len(streamlit.session_state['selected']) == 1:
+        file = streamlit.session_state['selected'][0]
+        metrics_of_file = report.get_metrics_source(chip, step, index)
+        file = file[file.rfind(f'{step}/{index}/') + len(f'{step}/{index}/'):]
+        if file in metrics_of_file:
+            metrics = ", ".join(metrics_of_file[file]) + "."
+            streamlit.success("This file includes the metrics of " + metrics)
+        else:
+            streamlit.warning("This file does not include any metrics.")
+
+
+def show_manifest(manifest):
+    """
+    Displays the manifest and a way to search through the manifest
+
+    Args:
+        manifest (dict) : represents the manifest json
+    """
+    streamlit.header('Manifest Tree')
+
+    search_cols = streamlit.columns([0.5, 0.5], gap="large")
+    key_search_col, value_search_col = search_cols
+
+    with key_search_col:
+        key = streamlit.text_input('Search Keys', '', placeholder="Keys")
+        if key != '':
+            manifest = report.search_manifest(manifest, key_search=key)
+
+    with value_search_col:
+        value = streamlit.text_input('Search Values', '', placeholder="Values")
+        if value != '':
+            manifest = report.search_manifest(manifest, value_search=value)
+
+    numOfKeys = report.get_total_manifest_parameter_count(manifest)
```

Comment:
`expand = report.get_total_manifest_parameter_count(manifest) < 20`
maybe make the 20 an input with a default